packages/loader/lib/index.js

```diff
@@ -1,12 +1,32 @@
-/*globals process */
-let define, require, Ember;
-
-// Used in @ember/-internals/environment/lib/global.js
-mainContext = this; // eslint-disable-line no-undef
+/* eslint-disable no-var */
+/* globals global globalThis self */
+var define, require;
 
 (function () {
-  let registry;
-  let seen;
+  var globalObj =
+    typeof globalThis !== 'undefined'
+      ? globalThis
+      : typeof self !== 'undefined'
+      ? self
+      : typeof window !== 'undefined'
+      ? window
+      : typeof global !== 'undefined'
+      ? global
+      : null;
+
+  if (globalObj === null) {
+    throw new Error('unable to locate global object');
+  }
+
+  if (typeof globalObj.define === 'function' && typeof globalObj.require === 'function') {
+    define = globalObj.define;
+    require = globalObj.require;
+
```

Comment:
I think we might need to accommodate the no conflict APIs for `loader.js` here

https://github.com/ember-cli/loader.js#no-conflict

Since `loader` doesn't expose `loader.require`/`loader.define` or a way to access what the "new" aliases are we can't do this just yet.